docs: Add Contributing documentation (#218)
adevinta/spark-android: docs/CONTRIBUTING.md

While we currently don't accept issues from people outside the Adevinta Platform to this repo, we
welcome [discussions and questions here](https://github.com/adevinta/spark-android/discussions/categories/general)!

## If you're part of the Adevinta Platform ⬇️

### Requirements

#### [git-lfs](https://git-lfs.com/)

This repository uses git-lfs to handle [cashapp/paparazzi](https://github.com/cashapp/paparazzi)
screenshot testing.  
If you introduce visible changes, you'll likely have to update screenshots files for the tests to
pass.

```bash
# Install
brew install git-lfs
# Check
git lfs install
```

#### [Java 17](https://github.com/adevinta/spark-android/issues/74)

We currently use **AGP 8.1.0** which requires developers to use **JDK 17** on Gradle JDK.

_If you're on macOS, you can install it with [brew](https://formulae.brew.sh/formula/openjdk@17)_

```bash
brew install openjdk@17
```

> ℹ️ If you're using a device with [Apple silicon (M1/M2)](https://support.apple.com/en-us/HT211814) then you might need to install
> a [zulu distribution](https://www.azul.com/downloads/zulu-community/?version=java-17-lts&architecture=x86-64-bit&package=jdk)

> or install it directly from **Android Studio**:
> 
> ![](../art/docs/jdk.png)

Reviewer: This image shows the wrong version.
I think it would more more useful for users to only have the path of the corresponding setting.

```suggestion
or install it directly from **Android Studio**:  
`File` → `Settings` → `Build, Execution, Deployment` → `Build Tools` → `Gradle` → `Gradle JDK`
```

Author: Yeah, it's way better 👍

# Contributing

### Issues, Bugs, Feature or Component Requests

While we currently don't accept issues from people outside the Adevinta Platform to this repo, we
welcome [discussions and questions here](https://github.com/adevinta/spark-android/discussions/categories/general)!

## If you're part of the Adevinta Platform ⬇️

### Requirements

#### [git-lfs](https://git-lfs.com/)

This repository uses git-lfs to handle [cashapp/paparazzi](https://github.com/cashapp/paparazzi)
screenshot testing.  
If you introduce visible changes, you'll likely have to update screenshots files for the tests to
pass.

```bash
# Install
brew install git-lfs
# Check
git lfs install
```

#### [Java 17](https://github.com/adevinta/spark-android/issues/74)

We currently use **AGP 8.1.0** which requires developers to use **JDK 17** on Gradle JDK.

You can find the latest JDK 17
release [here](https://www.azul.com/downloads/zulu-community/?version=java-17-lts&architecture=x86-64-bit&package=jdk).

_If you're on macOS, you can install it with [brew](https://github.com/mdogan/homebrew-zulu/)_

```bash
brew tap mdogan/zulu
brew install zulu-jdk17

> ```

Reviewer: IMO we should not present this information like this.

1. Zulu is one of many distributions, and we should not enforce anyone to use it, since we do not depend on it.
2. We should not provide a link to a custom brew tap. Instead, we should provide a better default like
   ```bash
   brew install openjdk@17
   ```
   and mention that M1 (apple silicon chip) devices might need a zulu distribution and link to their website instead (or [this](https://formulae.brew.sh/cask/zulu#default))

# Contributing

### Issues, Bugs, Feature or Component Requests

> While we don't accept issues from people outside the Adevinta Platform to this repo currently, we

Reviewer: ```suggestion
While we currently don't accept issues from people outside the Adevinta Platform to this repo, we
```

welcome [discussions and questions here](https://github.com/adevinta/spark-android/discussions/categories/general)!

## If you're part of the Adevinta Platform ⬇️

### Requirements

#### [git-lfs](https://git-lfs.com/)

> This repository uses [cashapp/paparazzi](https://github.com/cashapp/paparazzi) for screenshot
> testing purposes.  
> If you introduce visible changes, you'll likely have to update screenshots files for the test to
> pass.

Reviewer: ```suggestion
This repository uses git-lfs to handle [cashapp/paparazzi](https://github.com/cashapp/paparazzi) screenshot
testing.  
If you introduce visible changes, you'll likely have to update screenshots files for the tests to
pass.
```

```bash
# Install
brew install git-lfs
# Check
git lfs install
```

> #### [java 17](https://github.com/adevinta/spark-android/issues/74)

Reviewer: ```suggestion
#### [Java 17](https://github.com/adevinta/spark-android/issues/74)
```

> We use AGP 8.1.0 which requires developers to use JDK 17 on Gradle JDK

Reviewer: ```suggestion
We currently use AGP 8.1.0 which requires developers to use JDK 17 on Gradle JDK.
```

> If you're on macOS, you can install it with [brew](https://github.com/mdogan/homebrew-zulu/)

Reviewer: I would not encourage people to install external brew taps. Instead we should redirect them to the official source of JDK zulu (and maybe add the link to mdogan's tap)

```bash
brew tap mdogan/zulu
brew install zulu-jdk17
```

or install it directly from Studio

[//]: # (https://files.slack.com/files-pri/T0182J7UXEH-F04RH9F5U2G/image.png)

> [//]: # (https://files.slack.com/files-pri/T0182J7UXEH-F04RTEHLJQH/image.png)

Reviewer: These are not valid Markdown links

### How do I make a proper pull request?

> - Fork the project on GitHub

Reviewer: ```suggestion
- [Fork](https://github.com/adevinta/spark-android/fork) the project
```

> - Create a new branch

Reviewer: This is not strictly necessary if they have a fork.

- Upload the changes to your branch
- Check the [style guidelines](https://github.com/akyoto/quality/blob/master/STYLE.md)
- Create a pull request

### What if I already made a pull request but want to add more commits?

If you correctly created a new branch for your changes, you can simply upload the new commits to
your fork and they will automatically appear in the PR.

## Pull Requests

The best way to make an impact is by creating code submissions called pull requests. Pull requests
should be ‘solutions’ to GitHub issues.

To make a pull request:

1. Make sure there’s a GitHub issue for the change you’re proposing.
2. [Fork](https://help.github.com/articles/fork-a-repo/) the repo for the platform your code works
   in.
3. Write code on a branch in your fork.
4. [Create a pull request](https://help.github.com/articles/creating-a-pull-request/) to merge your
   branch’s contributions into the corresponding Spark repo by following the .
5. The pull request will be triaged by a #spark-dev member and code reviewed by a
   #spark-contributors-android member.
6. If the pull request is accepted, the accepting #spark-dev member will merge the pull request for

>    you.

Reviewer: This feels redundant with the previous `### How do I make a proper pull request?`

We currently use **AGP 8.1.0** which requires developers to use **JDK 17** on Gradle JDK.

> _If you're on macOS, you can install it with [brew](https://github.com/mdogan/homebrew-zulu/)_

Reviewer: ```suggestion
_If you're on macOS, you can install it with [brew](https://formulae.brew.sh/formula/openjdk@17)_
```

```bash
brew install openjdk@17
```

> > ℹ️ If you're using a device with an apple silicon ship (M1/M2) then you may be interested on using

Reviewer: ```suggestion
> ℹ️ If you're using a device with [Apple silicon (M1/M2)](https://support.apple.com/en-us/HT211814) then you might need to install
```

🚢 ?
